### PR TITLE
cleanup

### PR DESCRIPTION
**Pull request by clementhk**
@Xenopathic @DeepDiver1975 

one less call to horde in guessBestMailBox()

getSpecialFolder() is always returning Mailbox[]

remove duplicated code as guessSpecialRole()

move comment to more proper place

make use of new getHordeMailBox()

### CODE REVIEW

**Comment by Xenopathic** [comment]
:+1: cc @DeepDiver1975

File: lib/account.php [hunk 1]
```diff
@@ -308,38 +307,28 @@ public function getSentFolder() {
...
-				$createTrash = false;
-			}
+			$hordeTrashMailBox = new Horde_Imap_Client_Mailbox('Trash');
+			$createTrash = true;
 		}
 
 		$hordeMessageIds = new Horde_Imap_Client_Ids($messageId);
-		$hordeTrashMailBox = new Horde_Imap_Client_Mailbox($trashId);
 
 		if ($sourceFolderId === $trashId) {
```
Comment (DeepDiver1975) [inline lib/account.php]: $trashId is undefined if I see this correctly

**Comment by clementhk** [comment]
You got me :) fixed.

**Comment by jancborchardt** [comment]
As the original pull was now merged into master, you should issue a new pull request based off master. ;)

**Comment by clementhk** [comment]
Well if you guys want this then feel free to pull from my `clementhk:cleanup-2` branch.

**Comment by jancborchardt** [comment]
@clementhk can you issue a new pull request to master?

**Pull request closed without merging**